doc/source/ray-air/examples/convert_existing_pytorch_code_to_ray_air.md

```diff
@@ -0,0 +1,607 @@
+    model.eval()
+    test_loss, correct = 0, 0
+    with torch.no_grad():
+        for X, y in dataloader:
+            X, y = X.to(device), y.to(device)
+            pred = model(X)
+            test_loss += loss_fn(pred, y).item()
+            correct += (pred.argmax(1) == y).type(torch.float).sum().item()
+    test_loss /= num_batches
+    correct /= size
+    # print(f"Test Error: \n Accuracy: {(100*correct):>0.1f}%, Avg loss: {test_loss:>8f} \n")
+    return test_loss
+```
+
+And lastly, the wrapping `train_func()` where we added 4 lines and modified 2 (apart from the config dict):
+
+```{code-cell} ipython3
+import ray.train as train
+
+
+def train_func(config: dict):
+    batch_size = config["batch_size"]
+    lr = config["lr"]
+    epochs = config["epochs"]
+    
+    batch_size_per_worker = batch_size // train.world_size()
+    
+    # Create data loaders.
+    train_dataloader = DataLoader(training_data, batch_size=batch_size_per_worker)
+    test_dataloader = DataLoader(test_data, batch_size=batch_size_per_worker)
+    
+    train_dataloader = train.torch.prepare_data_loader(train_dataloader)
+    test_dataloader = train.torch.prepare_data_loader(test_dataloader)
+    
+    model = NeuralNetwork()
+    model = train.torch.prepare_model(model)
+    
+    loss_fn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=lr)
+    
+    for t in range(epochs):
+        train_epoch(train_dataloader, model, loss_fn, optimizer)
+        test_loss = test_epoch(test_dataloader, model, loss_fn)
+        train.report(loss=test_loss)
+
+    print("Done!")
+```
+
+Now we'll use Ray Train's TorchTrainer to kick off the training. Note that we can set the hyperparmameters here! In the `scaling_config` we can also configure how many parallel workers to use and if we want to enable GPU training or not.
+
+```{code-cell} ipython3
+from ray.train.torch import TorchTrainer
+
+
+trainer = TorchTrainer(
+    train_loop_per_worker=train_func,
+    train_loop_config={"lr": 1e-3, "batch_size": 64, "epochs": 4},
+    scaling_config={"num_workers": 2, "use_gpu": False},
+)
+result = trainer.fit()
+print(f"Last result: {result.metrics}")
+```
+
+Great, this works! You're now training your model in parallel. You could now scale this up to more nodes and workers on your Ray cluster.
+
+But there are a few improvements we can make to the code in order to get the most of the system. For one, we should enable **checkpointing** to get access to the trained model afterwards. Additionally, we should optimize the **data loading** to take place within the workers.
+
++++
+
+### Enabling checkpointing to retrieve the model
+Enabling checkpointing is pretty easy - we just need to call the `train.save_checkpoint()` API and pass the model state to it:
+
+```python
+    train.save_checkpoint(epoch=t, model=model.module.state_dict())
+```
+
+Note that the `model.module` part is needed because the model gets wrapped in `torch.nn.DistributedDataParallel`.
```

Comment:
```suggestion
Note that the `model.module` part is needed because the model gets wrapped in `torch.nn.DistributedDataParallel` by `train.torch.prepare_model`.
```